modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -809,12 +810,13 @@ private static Response prepareRamIndex(
                     BytesReference document = request.contextSetup.document;
                     XContentType xContentType = request.contextSetup.xContentType;
                     String id;
+                    SourceToParse sourceToParse;
                     if (indexService.getIndexSettings().getMode() == IndexMode.TIME_SERIES) {
-                        id = null; // The id gets auto generated for time series indices.
+                        sourceToParse = new SourceToParse(null, document, xContentType, TimeSeriesRoutingHashFieldMapper.encode(0));
```

Comment:
Maybe put `TimeSeriesRoutingHashFieldMapper.encode(0)` into a constant in `TimeSeriesRoutingHashFieldMapper` c;ass and reuse that places where we need a dummy routing?

Reply:
Done.